Message.hpp

```diff
@@ -0,0 +1,49 @@
+#ifndef MESSAGE_HPP
+#define MESSAGE_HPP
+
+#include <string>
+#include <cstdint>
+#include <ctime>
+#include <iomanip>
+#include <sstream>
+#include <cstring>
+
+/**
+ * @class Message
+ * @brief Represents a message with a size, timestamp, and content.
+ *
+ * This class provides functionality to store a message along with its size
+ * and timestamp, and to convert the message into a formatted string.
+ */
+class Message {
+public:
+    /**
+     * @brief Constructs a Message object.
+     *
+     * @param size The size of the message content in bytes.
+     * @param timestamp The timestamp of when the message was created, in Unix time.
+     * @param msg The content of the message.
+     */
+    Message(std::uint32_t size, std::uint64_t timestamp, const std::string& msg);
+
+    /**
+     * @brief Converts the message to a formatted string.
+     *
+     * @return A string representation of the message in the format "size$HH:MM$message".
+     */
+    std::string toString() const;
+
+private:
+    std::uint32_t _size;           ///< The size of the message content in bytes.
+    std::uint64_t _timestamp;      ///< The timestamp of the message, in Unix time.
+    static constexpr uint32_t BUFFER = 4025; ///< The maximum buffer size for message content.
```

Comment:
for constants i want you to use `auto`
for example
`static constexpr auto COLOR = "red"`;
or
`static constexpr auto NUMBER = 5;`

auto makes the compiler deduce a parameter's type during compilation